File: Семестр 1/ДЗ 8/list.cpp
#include "list.h"
#include <iostream>
using namespace std;

List *createList()
{
	List *list = new List;
	list->head = new ListElement;
	list->head->next = NULL;
	list->head->value = ""; 
	list->head->count = 0;
	return list;
}

Position elemPos(char* value, List *list)
{
	ListElement* tmp = list->head;
	while(strcmp(tmp->value, value) != 0)
		tmp=tmp->next;
	return tmp;
}

void addElem(Position p, char* value)
{
	ListElement* tmp = new ListElement;
	tmp->value = new char[100];
	tmp->value = value;
	tmp->next = p->next;
	p->next = tmp;
}

void deleteElem(char* value, List *list)
{
	ListElement* tmp = list->head;
	while (strcmp(tmp->next->value, value) != 0)
		tmp = tmp->next;
	ListElement *tmp1 = elemPos(value, list);
	tmp->next = tmp1->next;
    delete tmp1;
}

void printList(List *list)
{
	ListElement* tmp = list->head;
	while (tmp->next != NULL)
	{
		cout << tmp->next->value << " " << tmp->next->count << endl << endl;
		tmp = tmp->next;
	}
}

void deleteList(List *list)
{
	ListElement *tmp = list->head;
	ListElement *tmp1 = tmp->next;
	while (tmp1 != NULL)
	{
		tmp = tmp1;
		tmp1 = tmp1->next;
		delete tmp;
	}
	delete list->head;
	delete list;
}

bool isEmpty(List *list)
{
	return list->head->next == NULL;
}

File: Семестр 2/ДЗ 5/problem_2/problem_2/List.cs
using System;

namespace problem_2
{
    public class List
    {
        public class ListElement
        {
            public int Number { get; set; }
            public ListElement Next { get; set; }
        }
        public ListElement Head { get; set; }

        public List()
        {
            Head = new ListElement();
            Head.Next = null;
            Head.Number = 0;
        }

        public bool Exists(int elem)
        {
            var tmp = Head.Next;
            while (tmp != null)
            {
                if (tmp.Number == elem)
                {
                    return true;
                }
                tmp = tmp.Next;
            }
            return false;
        }

        public int SizeOfList()
        {
            int size = 0;
            ListElement tmp = Head.Next;
            while (tmp != null)
            {
                ++size;
                tmp = tmp.Next;
            }
            return size;
        }

        public ListElement ElemPos(int value)
        {
            ListElement tmp = Head;
            while (tmp.Number != value)
                tmp = tmp.Next;
            return tmp;
        }

        public virtual void AddElem(int elem)
        {
            var tmp = Head;
            while (tmp.Next != null)
                tmp = tmp.Next;
            AddElem(tmp, elem);
        }

        public void AddElem(ListElement pos, int value)
        {
	        ListElement tmp = new ListElement();
	        tmp.Number = value;
	        tmp.Next = pos.Next;
	        pos.Next = tmp;
        }

        public virtual void DeleteElem(int value)
        {
            ListElement tmp = Head;
            while (tmp.Next.Number != value)
            {
                tmp = tmp.Next;
                if (tmp.Next == null)
                    break;
            }
            if (tmp.Next != null)
            {
                ListElement tmp1 = ElemPos(value);
                tmp.Next = tmp1.Next;  
            }          
        }

        public void PrintList()
        {
            ListElement tmp = Head;
            while (tmp.Next != null)
            {
                Console.Write("{0} ", tmp.Next.Number);
                tmp = tmp.Next;
            }
            Console.WriteLine();
        }
    }
}
